hoomd_organics/library/forcefields.py

```diff
@@ -187,3 +188,272 @@ def _create_forcefield(self):
+    from files storing the tabulated energy and forces.
+
+    In HOOMD-Blue, table potentials are available for:
+
+        * Pairs: `hoomd.md.pair.Table`
+        * Bonds: `hoomd.md.bond.Table`
+        * Angles: `hoomd.md.angle.Table`
+        * Dihedrals: `hoomd.md.dihedral.Table`
+
+    Notes
+    -----
+    HOOMD table potentials are initialized using arrays of energy and forces.
+    It may be most convenient to store tabulated data in files,
+    in that case use the `from_files` method.
+
+
+    Parameters
+    ----------
+    pairs: dict, optional, default None
+    bonds: dict, optional, default None
+    angles: dict, optional, default None
+    dihedrals: dict, optional, default None
+
+    """
+
+    def __init__(
+        self,
+        pairs=None,
+        bonds=None,
+        angles=None,
+        dihedrals=None,
+        r_min=None,
+        r_cut=None,
+        exclusions=["bond", "1-3"],
+        nlist_buffer=0.40,
+    ):
+        self.pairs = pairs
+        self.bonds = bonds
+        self.angles = angles
+        self.dihedrals = dihedrals
+        self.r_min = r_min
+        self.r_cut = r_cut
+        self.exclusions = exclusions
+        self.nlist_buffer = nlist_buffer
+        self.bond_width, self.angle_width, self.dih_width = self._check_widths()
+        self.hoomd_forcefield = self._create_forcefield()
+
+    @classmethod
+    def from_files(
+        cls,
+        pairs=None,
+        bonds=None,
+        angles=None,
+        dihedrals=None,
+        exclusions=["bond", "1-3"],
+        nlist_buffer=0.40,
+    ):
+        """Create table forefield using a `type: file_path` mapping.
+
+        Notes
+        -----
+        The parameters must use a `{"type": "file_path"}` mapping.
+        Following HOOMD conventions, pair types must be given as a tuple
+        of `("type1", "type2")` while bonds, angles and dihedrals
+        are givne as strings of `"type1-type2-type3"`
+
+        Parameters
+        ----------
+        pairs: dict, optional, default None
+            Dictionary of pair type values with file path keys
+        bonds: dict, optional, default None
+            Dictionary with keys of bond type and keys of file path
+        angles: dict, optional, default None
+            Dictionary with keys of angle type and keys of file path
+        dihedrals: dict, optional, default None
+            Dictionary with keys of dihedral type and keys of file path
+
+        Example
+        -------
+        .. code-block:: python
+
+            table_forcefield = TableForcefield.from_files(
+                pairs = {
+                    ("A", "A"): "A_pairs.txt
+                    ("B", "B"): "B_pairs.txt
+                    ("A", "B"): "AB_pairs.txt
+                },
+                bonds = {"A-A": "A_bonds.txt", "B-B": "B_bonds.txt"},
+                angles = {"A-A-A": "A_angles.txt", "B-B-B": "B_angles.txt"},
+            )
+        """
+
+        def _load_file(file):
```

Comment:
I wonder if we need a way to pass in np kwargs here by adding a kwargs param to the class.

Reply:
what do you mean by np kwargs?